app/backend/src/couchers/servicers/requests.py

```diff
@@ -116,6 +116,14 @@ def CreateHostRequest(self, request, context):
             host_request.from_last_seen_message_id = message.id
             session.add(host_request)
 
+            host_request = (
+                session.query(HostRequest)
+                .filter(HostRequest.conversation_id == conversation.id)
+                .filter(or_(HostRequest.from_user_id == context.user_id, HostRequest.to_user_id == host.id))
+                .one_or_none()
+            )
```

Comment:
You don't need to grab the host request from the database again, SQLAlchemy (SA) keeps track of it and you can use the one from above.

However, at this point, SA hasn't sent it to the database yet, so it's just an object in memory and so has no auto-generated ID column yet. You need to do a `session.flush()` so SA will flush it to the database returning back an ID (tracked by SA automatically). It's a good idea to flush before sending the email anyway, that way if for example a DB constraint is violated, the function will fail before the email gets sent out. (On failure the database transaction will `ROLLBACK` so we get that ACID goodness and no gibberish data is stored in the databse).

(The `with session_scope ...` auto-commits on exit, that's why there's few `session.commit()`s around in the code.)

[Sorry for the verbose comment :P.]